Services/Music_Portal.Services.Services/ArtistService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Domain.Core;
+using Infrastructure.Data;
+using Services.Interfaces;
+
+namespace Services.Services
+{
+    public class ArtistService : IArtistService
+    {
+        private readonly ILastFmService _lastFmService;
+        private readonly IMapper _mapper;
+        private readonly ApplicationContext _db;
+
+        public ArtistService(ILastFmService lastFmService, IMapper mapper, ApplicationContext db)
+        {
+            _lastFmService = lastFmService;
+            _mapper = mapper;
+            _db = db;
+        }
+
+        public async Task<IEnumerable<Artist>> GetTopArtists()
```

Comment:
here you should try to get top artists from database first, then if there are no exist, get from lastfm api

Reply:
But I make a request to API first only to update info about artists. I did it in case of clearer info.

Reply:
Let's use database info first, then we can add a job to update info if it will be needed, but now try to get info with database and then with LastFM API